cassandra/metadata.py

```diff
@@ -1435,7 +1435,11 @@ def as_cql_query(self):
                 index_target,
                 class_name)
             if options:
-                ret += " WITH OPTIONS = %s" % Encoder().cql_encode_all_types(options)
+                opts_cql_encoded = _encoder.cql_encode_all_types(options)
+                # PYTHON-1008
+                if isinstance(opts_cql_encoded, six.binary_type):
+                    opts_cql_encoded = opts_cql_encoded.decode('utf-8')
+                ret += " WITH OPTIONS = %s" % opts_cql_encoded
```

Comment:
I think I have the same question than before, shouldn't encoders deal with that, rather than adding a hack-decoding in metadata.py? If there is no reason `options` (or another cql parts) to be encoded directly to `binary_type`.. we should probably create a new encoder function that is aware of this and always returns unicode.

Reply:
Ah, I understand your objection now. If the `cql_encode_*` methods were only used for generating CQL for display, I'd be firmly +1. However, these methods are also used internally for binding arguments to non-prepared statements, and I think this use case wants `six.binary_type` instance -- I'll have a look at that now to confirm.

Are you proposing an interface that always returns a `six.string_type` or one that allows callers to choose?

Reply:
See [`bind_params`](https://github.com/datastax/python-driver/blob/850f218b5cde1fd000338517372395f690097153/cassandra/query.py#L891), [used here, for example](https://github.com/datastax/python-driver/blob/master/cassandra/cluster.py#L2239)